### PR TITLE
Xsd support for startup services

### PR DESCRIPTION
**Pull request by sunil-indoria**
This repo contains change
1. XSD for StartupServices model file
2. Generate cs file from above XSD
3. Create a dll with above model cs file
4. Include the dll and xsd in SF Internal nuget

### CODE REVIEW

**Comment by sunil-indoria** [comment]
Need to see how to fix xsd.exe command issue

**Review by PavanKunapareddyMSFT — APPROVED** [review]

File: properties/service_fabric_managed_netframework.targets [hunk 1]
```diff
@@ -50,6 +51,8 @@
   <!-- Copy prod  binaries for full net framework-->
   <Target Name="CopyProdBinaries_netframework">
     <ItemGroup>
+	    <BinairesNetFramework Include="src\Microsoft.ServiceFabric.StartupServicesUtility\$(OutputPath)WindowsFabricServiceModel\StartupServicesModel.xsd"/>
```
Comment (anantshankar17) [inline properties/service_fabric_managed_netframework.targets:54]: Rename WindowsFabricServiceModel -> StartupServicesModel, as this model has nothing to do with WinfabModelServiceModel

File: properties/service_fabric_managed_netstandard.targets [hunk 2]
```diff
@@ -29,6 +30,8 @@
     <Exec Command="dotnet publish &quot;$(RepoRoot)src\netstandard\FabActUtil\FabActUtil_netstandard.csproj&quot; -o &quot;$(OutputPath)Publish&quot; --no-restore --nologo" />
 
     <ItemGroup>
+	    <BinairesNetStandard Include="$(RepoRoot)src\netstandard\Microsoft.ServiceFabric.StartupServicesUtility\$(OutputPath)WindowsFabricServiceModel\StartupServicesModel.xsd"/>
```
Comment (anantshankar17) [inline properties/service_fabric_managed_netstandard.targets:33]: same comment as above

File: properties/service_fabric_nuget.targets [hunk 3]
```diff
@@ -23,6 +24,7 @@
     <Message Text="Copying NuGet packages to drop folder ..." Importance="High"/>
     <ItemGroup>
       <Nupkgs Include="nuprojs\Microsoft.ServiceFabric.Actors\$(OutputPath)\Microsoft.ServiceFabric.Actors.*.nupkg"/>
+	    <Nupkgs Include="nuprojs\Microsoft.ServiceFabric.StartupServicesUtility\$(OutputPath)\Microsoft.ServiceFabric.StartupServicesUtility.*.nupkg"/>
```
Comment (anantshankar17) [inline properties/service_fabric_nuget.targets:27]: are you creating a new nuget ? I thought the xsd/dll were to be packaged in the SF_Internal nuget.

File: src/Microsoft.ServiceFabric.StartupServicesUtility/Microsoft.ServiceFabric.StartupServicesUtility.csproj [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  </ItemGroup>
+
+  <Target Name="GenerateCsFromXsd" BeforeTargets="CoreCompile" Outputs="$(OutputPath)\WindowsFabricServiceModel\StartupServicesModel.cs" Inputs="$(XsdModelPath)\StartupServicesModel.xsd">
+    <Message Text="Building StartupServicesModel.cs using XSD.exe" />
+    <Exec Command="$(XsdPath)\xsd.exe /c /o:$(OutputPath)WindowsFabricServiceModel /n:Microsoft.ServiceFabric.StartupServicesUtility $(OutputPath)\WindowsFabricServiceModel\StartupServicesModel.xsd" />
+  </Target>
+
+  <Target Name="CopyXsdFile" Outputs="$(OutputPath)WindowsFabricServiceModel\StartupServicesModel.xsd" Inputs="$(XsdModelPath)\StartupServicesModel.xsd" BeforeTargets="GenerateCsFromXsd">
+    <MakeDir Directories="$(OutputPath)\WindowsFabricServiceModel" />
+    <Copy SourceFiles="$(XsdModelPath)\StartupServicesModel.xsd" DestinationFolder="$(OutputPath)WindowsFabricServiceModel" SkipUnchangedFiles="true" />
```
Comment (anantshankar17) [inline src/Microsoft.ServiceFabric.StartupServicesUtility/Microsoft.ServiceFabric.StartupServicesUtility.csproj:29]: rename out folder name aswell.

File: src/netstandard/Microsoft.ServiceFabric.StartupServicesUtility/Microsoft.ServiceFabric.StartupServicesUtility_netstandard.csproj [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  <ItemGroup>
+    <Compile Include="$(OutputPath)\WindowsFabricServiceModel\StartupServicesModel.cs"/>
+  </ItemGroup>
+
+<Target Name="GenerateCsFromXsd" BeforeTargets="CoreCompile" Outputs="$(OutputPath)\WindowsFabricServiceModel\StartupServicesModel.cs" Inputs="$(XsdModelPath)\StartupServicesModel.xsd">
+    <Message Text="Building StartupServicesModel.cs using XSD.exe" />
+    <Exec Command="$(XsdPath)\xsd.exe /c /o:$(OutputPath)WindowsFabricServiceModel /n:Microsoft.ServiceFabric.StartupServicesUtility $(OutputPath)\WindowsFabricServiceModel\StartupServicesModel.xsd" />
+  </Target>
+
+  <Target Name="CopyXsdFile" Outputs="$(OutputPath)WindowsFabricServiceModel\StartupServicesModel.xsd" Inputs="$(XsdModelPath)\StartupServicesModel.xsd" BeforeTargets="GenerateCsFromXsd">
```
Comment (anantshankar17) [inline src/netstandard/Microsoft.ServiceFabric.StartupServicesUtility/Microsoft.ServiceFabric.StartupServicesUtility_netstandard.csproj:25]: same comment as above.

**Review by anantshankar17 — CHANGES_REQUESTED** [review]
the folder path is confusing "xsd/xsd.exe". Should be something like: "CodeGenerationTool/xsd.exe"

File: properties/service_fabric_nuget.targets (hunk 3 above)
Comment (sunil-indoria) [reply properties/service_fabric_nuget.targets:27]: xsd and dll are packed in SF_Internal nuget. Along with this there is a separate nuget as well which can be used if someone needs only this nuget.

Comment (anantshankar17) [reply properties/service_fabric_nuget.targets:27]: Who would need this dll and xsd apart from the WF-SDK for VS  ? This XSD is only meant to be used with VS.

Comment (sunil-indoria) [reply properties/service_fabric_nuget.targets:27]: Taking this out from nuget.

File: properties/service_fabric_managed_netframework.targets (hunk 1 above)
Comment (sunil-indoria) [reply properties/service_fabric_managed_netframework.targets]: Fixed.

File: properties/service_fabric_managed_netstandard.targets (hunk 2 above)
Comment (sunil-indoria) [reply properties/service_fabric_managed_netstandard.targets]: Fixed

File: properties/service_fabric_nuget.targets (hunk 3 above)
Comment (sunil-indoria) [reply properties/service_fabric_nuget.targets]: Fixed

File: src/netstandard/Microsoft.ServiceFabric.StartupServicesUtility/Microsoft.ServiceFabric.StartupServicesUtility_netstandard.csproj (hunk 5 above)
Comment (sunil-indoria) [reply src/netstandard/Microsoft.ServiceFabric.StartupServicesUtility/Microsoft.ServiceFabric.StartupServicesUtility_netstandard.csproj]: Fixed

**Review by anantshankar17 — APPROVED** [review]

**Pull request merged**